src/ic_renderer.h

```diff
@@ -23,18 +17,27 @@ namespace IC::Renderer
 
     class Renderer
     {
-    protected:
-        GLFWwindow *Window;
-
-        static Renderer *MakeVulkan(RendererConfig &rendererConfig);
-
     public:
         Renderer(RendererConfig config) { Window = config.Window; };
         ~Renderer() = default;
 
         virtual void DrawFrame() = 0;
         virtual void AddMesh(Mesh &meshData, Material &materialData) = 0;
 
+    protected:
+        GLFWwindow *Window;
+
+    private:
+#ifdef IC_RENDERER_VULKAN
+        static Renderer *MakeVulkan(RendererConfig &rendererConfig);
+#else
+        static Renderer *MakeVulkan(RendererConfig &rendererConfig)
+        {
+            return nullptr;
+        }
+#endif
+
+    public:
         static Renderer *MakeRenderer(RendererConfig &rendererConfig)
```

Comment:
Can we add a corresponding DestroyRenderer function?

Reply:
I think this is lifted from how [blah](https://github.com/NoelFB/blah) does it, where `MakeRenderer` is called, then just `delete`'d later. I'll leave this to @hillj35 though.